pkg/storage/redis.go

```diff
@@ -0,0 +1,366 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	goredislib "github.com/go-redis/redis/v8"
+	"github.com/go-redsync/redsync/v4"
+	"github.com/go-redsync/redsync/v4/redis/goredis/v8"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	PONG               = "PONG"
+	RedisScanBatchSize = 1000
+	RedisMutex         = "redis-mutex"
+)
+
+func init() {
+	err := RegisterStorage(&RedisDB{})
+	if err != nil {
+		panic(err)
+	}
+}
+
+type RedisDB struct {
+	db    *goredislib.Client
+	ctx   context.Context
+	mutex *redsync.Mutex
+}
+
+func (b *RedisDB) Init(i interface{}) error {
+	options := i.(map[string]interface{})
+
+	client := goredislib.NewClient(&goredislib.Options{
+		Addr:     options["address"].(string),
+		Password: options["password"].(string),
+	})
+
+	pool := goredis.NewPool(client)
+	rs := redsync.New(pool)
+
+	b.db = client
+	b.mutex = rs.NewMutex(RedisMutex)
+	b.ctx = context.Background()
+
+	if options["flush"] != nil && options["flush"].(bool) {
+		err := b.db.FlushAll(b.ctx).Err()
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
This seems a bit dangerous, no? If this is for testing, perhaps there is a way of preventing it from happening in production? I can see build tags being used for such a case.

Reply:
I'll take this flush option out and have it flush in our docker compose